sos/plugins/npm.py

```diff
@@ -85,6 +85,7 @@ def _find_modules_in_npm_cache(self):
         self.archive.add_string(json.dumps(output), outfn)
 
     def setup(self):
+        project_path = None
```

Comment:
This doesn't really work (nicely):

* It fixes the bug reported in #951 since `project_path` is always assigned a value

BUT:

* The 1st `npm config list` outside the `if self.get_option("project_path") (line 94) will now misbehave
  - Receives `working_directory=None`, which translates to `runat=None` - which is fine, but it makes it identical to the system `npm config list` on line 99 (except it's called `npm_config_list_project`). That's potentially confusing for users and people analysing the report.

So either the `add_cmd_output()` that uses `project_path` must move inside that `if` (in which case there's no need to assign `project_path = None` outside the branch as it's never used outside a branch where its value is assigned), or `project_path` has to be something valid to be passed to that `npm config list` call.

The first seems more straightforward.